package-res/resources/web/vizapi/DataTable.js

```diff
@@ -15,1094 +15,1220 @@
+  return this;
+};
 
-    columnIndex The index of the column to set the property on
-    name        The name of the property
-    value       The value of the property
-*/
-pentaho.DataView.prototype.setColumnProperty = function(columnIndex, name, value) {
-    this.dataTable.setColumnProperty(columnIndex, name, value);
-}
+/**
+ * Gets the indexes of the visible source rows.
+ *
+ * Do **NOT** modify the returned array directly.
+ *
+ * @method getViewRows
+ * @return {Array|Null} The array of source row indexes or `null`, if one has not been set.
+ * @since 3.0
+ */
+pentaho.DataView.prototype.getViewRows = function() {
+  return this._rows;
+};
 
-/*
-    getColumnProperty
-    Returns a column property
+// -----
+// TABLE
 
-    columnIndex The index of the column to set the property on
-    name        The name of the property
+/*
+ * Gets the view's source table.
+ *
+ * @method getSourceTable
+ * @return {AbstractDataTable} The source table.
+ * @since 3.0
+ */
+pentaho.DataView.prototype.getSourceTable = function() {
+  return this._source;
+};
 
-    Return      The value of the property
+/**
+ * Gets a {{#crossLink "DataTable"}}{{/crossLink}} loaded with the
+ * data view's visible rows and columns.
+ *
+ * This is a convenience method for abbreviating:
+ *
+ *      var dataTable = new pentaho.DataTable(thisDataView.toJSON());
+ *
+ * @method toDataTable
+ * @return {DataTable} A data table.
 */
-pentaho.DataView.prototype.getColumnProperty = function(columnIndex, name) {
-    return this.dataTable.getColumnProperty(columnIndex, name);
+pentaho.DataView.prototype.toDataTable = function() {
+  return new pentaho.DataTable(this.toJSON());
 };
 
 /* TRENDS */
 
-(function(){
+(function() {
 
-    function argRequired(name){
-        return new Error("Argument '" + name + "' is required.");
+    function argRequired(name) {
+      return new Error("Argument '" + name + "' is required.");
     }
-    
-    function argInvalid(name, text){
-        return new Error("Argument '" + name + "' is invalid." + (text ? (" " + text) : ""));
+
+    function argInvalid(name, text) {
+      return new Error("Argument '" + name + "' is invalid." + (text ? (" " + text) : ""));
     }
-    
-    /* createTrend
-     * Computes a trend of a given type and adds the 
-     * result to a new column in the data table.
-     *  
-     * trendArgs Keyword arguments
-     * trendArgs.type  The type of trend to create; possible values: 'linear'
-     * trendArgs.x     The index of the "x" value column; can be a numeric or string column
-     * trendArgs.y     The index of the "y" value column; must be of a column of type 'number'
-     * trendArgs.name  The name of the new trend column; defaults to the type of trend plus the suffix "Trend"
-     * trendArgs.label The label of the new trend column; defaults to the trend name, if specified, or the default label of the trend type.
-     * 
+
+    // TODO: Needs unit-testing before being publicly documented...
```

Comment:
This part of the code was not included in the story "budget", so we won't document and unit test it now. There's an issue for enhancing docs and unit tests that fit this bill.
The comment adverts developers not to make the doclet visible (by adding the second "*").